src/main/java/io/nats/client/impl/NatsConnectionWriter.java

```diff
@@ -49,10 +50,13 @@
         this.stopped = new CompletableFuture<>();
         ((CompletableFuture<Boolean>)this.stopped).complete(Boolean.TRUE); // we are stopped on creation
 
-        this.sendBuffer = new byte[connection.getOptions().getBufferSize()];
+        int bufSize = connection.getOptions().getBufferSize();
+        this.sendBuffer = new byte[bufSize];
 
-        outgoing = new MessageQueue(true);
-        reconnectOutgoing = new MessageQueue(true);
+        // Limit the messages that we can push on the queue to 5x the number we would send in a single batch
+        this.maxAccumulate = 1000;

```

Comment:
Should this be based on total pending size in bytes vs msgs?

Reply:
ultimately it is limited by what is there, but this is a fixed value to prevent trying to read too much.

Reply:
Read or Write?

Reply:
write